### DISCUSSION TITLE
Why I will not answer every question anymore...

### DISCUSSION THREAD

**Discussion by rdehuyss** [jobrunr/jobrunr#769] [General]
## TLDR;
Open-source development is hard and takes a lot of my (free) time which is not being paid for. **If your question is related to your setup or to some feature covered or solved in JobRunr Pro, I will not answer it anymore** - this to prevent burnout and to make sure that I can spend more quality time with my kids and great wife. More [here](#user-content-i-wont-answer-every-question).

<br />

## Big shoutout to my existing customers 🙏  🥳  ❤️
First of all, I want to take this moment to thanks all the existing JobRunr Pro customers - without them JobRunr would not be what it is today. This also includes all the **developers that stick out their neck** to ask their manager / CTO for a JobRunr Pro license. 
So a big 🙏 for your continued support and please know that I really appreciate you for that - you have my eternal ❤️. In fact, it is **thanks to you** that I can keep doing what I love to do - improving JobRunr and JobRunr Pro.

<br />

## Open-source is the digital infrastructure of today's society
> Our modern society—everything from hospitals to stock markets to newspapers to social media—runs on software. But take a closer look, and you’ll find that **the tools we use to build software are buckling under demand**. Nearly all software today relies on free, open-source code, written and maintained by communities of developers and other talent. Much like roads or bridges, which anyone can walk or drive on, open-source code can be used by anyone—from companies to individuals—to build software. This type of code makes up the digital infrastructure of our society today.
> 
> Just like physical infrastructure, **digital infrastructure needs regular upkeep and maintenance**. In the United States, over halfof government spending on transportation and water infrastructure goes just to maintenance. But financial support for digital infrastructure is much harder to come by.
<p align="right">from <a href="https://www.fordfoundation.org/media/2976/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure.pdf"><strong>Roads and Bridges</strong> - the Unseen Labor Behind Our Digital Infrastructure</a> by Nadia Eghbal</p>

<br/>

## Open-source development is hard
Open source development is hard. Yes, it is simple to create a GitHub repo and open-source your code. However, if your project is successfully, the real work only starts then. You may not know this but open-source also leads to **burnout and depression** - and this is proven by many different papers <sup>[[1]](#user-content-ref-1) [[2]](#user-content-ref-2) [[3]](#user-content-ref-3) [[4]](#user-content-ref-4) [[5]](#user-content-ref-5) [[6]](#user-content-ref-6) [[7]](#user-content-ref-7) .

If users find your open-source software helpful (either for personal use or for professional use) they ask for support, create bug reports and ask for more features. As a responsible "padre de familia" you maintain your project and provide support, fix bugs and improve your project which results in more success, increased user adoption but again more users asking for support and features. It's a vicious circle that, like many open-source developers, you'll probably do during your free time which takes away time you can spend with your loved ones.

Each time a bug report is created in JobRunr, my heart still skips a beat - it means someone has troubles with the code I have written. Although JobRunr is only 2 years old, it has processed over **9 500 000 000 jobs** (yes - that's billion) in less than a year (or about **300 jobs per second**), and is covered with over 3000 tests, I still not dare to think it is bug free ([imposter syndrome](https://en.wikipedia.org/wiki/Impostor_syndrome)?) and I take every bug report serious. Even if I'm not behind the computer, there is almost always a background job running (pun intended 🤓) on how to improve JobRunr.

<br />

## Open-source != Free Software && Free Support
> Open source was/is not about free for the pure sake of lowering costs. [Free Software](https://www.gnu.org/philosophy/free-software-even-more-important.html) isn’t even really about $0, but rather zero restrictions.
FOSS was about the open creation of, contribution to, and use of software, because software has been and will continue to be so transformative for the entire world, it should not be restricted and allowed to be a proprietary weapon that is used to dominate.
<p align="right">from <a href="https://www.linkedin.com/pulse/open-source-has-failed-darin-bartik">Open Source Has NOT Failed</a> by Darin Bartik</p>

I know of a Fortune top 100 company that uses JobRunr - it made more money in 2022 then the amount of jobs JobRunr has processed 🤣, and yet I know it violates the AGPL license. Am I going to do anything about this? No, as I think we software developers are pacifists and I prefer not spend time & energy around this - I prefer to code. Would I appreciate them thanking me for my work by getting a Pro license? Definitely!

<br />

## Why I believe in open-source development
As I have been a software developer and architect for most of my life, I know how hard it is if a library doesn't behave the way you think it would. By having the source code of that library available, not only I have been able to find out why it was behaving differently - it also gave me lots of opportunities to **learn** and improve my skillset.

I also think that open-source is our **digital infrastructure (aka our roads and bridges of the internet age)** and we as humanity would not be in the same place as where we are today if we would not have had the whole open-source community.

On top of that, it's thanks to JobRunr being open-source that I've been invited to talk in different parts of the world about the thing I love to do - programming. And, if the growth of JobRunr continues like it is this last year, I hope to be break even by the end of the year.

<br />
<span id="i-wont-answer-every-question"></span>

## So, why I won't answer every question anymore
I won't help users anymore trying to work around problems that are solved in [JobRunr Pro](https://www.jobrunr.io/en/documentation/pro/). And there are many reasons for that:
-  JobRunr is open-source meaning that you can navigate through the source code right here on GitHub. In **each and every case**, you can find your answer in the code but this takes time. So, I ask you to choose - either **spend your own time** on finding an answer or take a [JobRunr Pro License](https://www.jobrunr.io/en/pricing/) where I will help you with pleasure.
- I see JobRunr being used as a mission-critical component in various applications from even Fortune top 100 companies. If JobRunr adds value to your business then why should the open-source components - that help you and your company succeed - continue be the result of **unpaid labor**?  
- As you can see from the references, many open-source projects we take for granted stop being maintained because of burnout. By convincing your manager or CTO, you help with continued development and maintenance of JobRunr. This allows me to continue the work on JobRunr as my day-to-day job instead of doing all of this during the weekend - decreasing the chance that this becomes another unmaintained open-source project (not that I'm thinking of stopping the maintenance of JobRunr at this time).
- I hope to continue supporting JobRunr for years to come but I prefer to avoid a burnout at any cost and I also want to enjoy time with my kids and my great wife. 

<br />

## What can you do?
Well, if you're enjoying JobRunr I would really love if you bring up this topic with your manager, especially if JobRunr plays an important part in your software architecture. I also would appreciate if you spread the word about JobRunr as by doing some research I found out that only 2 to 3% of the Java developers know JobRunr. More companies using it means more support but also hopefully more JobRunr Pro licenses sold.

The other thing I also like is feedback and help with the documentation and any of the issues. 

<br />

### References
<span id="ref-1">[1] [Roads and Bridges - the Unseen Labor Behind Our Digital Infrastructure](https://www.fordfoundation.org/media/2976/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure.pdf) by Nadia Eghbal - 2016-06-14</span>
<span id="ref-2">[2] [Why Open Source Failed](https://medium.com/@johnmark/why-open-source-failed-6cae5d6a9f6) by John Mark - 2018-7-30</span>
<span id="ref-3">[3] [Save Open Source, Save the World](https://medium.com/@johnmark/save-open-source-save-the-world-f2b0bf6ec630) by John Mark - 2018-12-20</span>
<span id="ref-4">[4] [Open Source Has Not Failed. Don't Cover Up Corporate Abuse of Open Source](https://dev.to/lethargilistic/open-source-has-not-failed-dont-cover-up-corporate-abuse-of-open-source-3ffe) by Michael MacTaggert - 2018-08-17</span>
<span id="ref-5">[5] [moment().endOf(‘term’)](https://medium.com/@timrwood/moment-endof-term-522d8965689) by Tim Wood - 2016-07-27</span>
<span id="ref-6">[6] [Why Modern Open Source Projects Fail](https://arxiv.org/abs/1707.02327) by Jailton Coelho and Marco Tulio Valente - 2017-07-07</span>
<span id="ref-7">[7] [Is this GitHub Project Maintained?](https://arxiv.org/abs/2003.04755) by Jailton Coelho, Marco Tulio Valente, Luciano Milen, Luciana L. Silva - 2020-03-09</span>

**Comment by bpoole6** [comment]
I think in the back of all of minds we are aware that somebody is dedicating their free time to making software that is generally available for public use. Its good to sometimes bring that to the forefront of our conscience.

Thanks for the write up, for creating jobrunr and especially maintaining jobrunr.
